CoreWebServer/src/au/gov/asd/tac/constellation/webserver/api/HelpServlet.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.webserver.api;
+
+import au.gov.asd.tac.constellation.webserver.WebServer.ConstellationHttpServlet;
+import au.gov.asd.tac.constellation.webserver.help.SphinxHelpDisplayer;
+import java.io.IOException;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ * A servlet that serves help files.
+ * <p>
+ * The files live in a zip file. Each request opens the zip file and reads the
+ * requested resource.
+ *
+ * @author algol
+ */
+@ServiceProvider(service = ConstellationHttpServlet.class)
+@WebServlet(
+        name = "HelpServer",
+        description = "HTML help server",
+        urlPatterns = {"/help/*"})
+public class HelpServlet extends ConstellationHttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(HelpServlet.class.getName());
+
+    private static final Map<String, String> MIME_TYPES = Map.of(
+            ".css", "text/css",
+            ".gif", "image/gif",
+            ".html", "text/html",
+            ".jpg", "image/jpeg",
+            ".js", "text/javascript",
+            ".png", "image/png",
+            ".txt", "text/plain"
+    );
+
+    @Override
+    protected void doGet(final HttpServletRequest request, final HttpServletResponse response) throws ServletException {
+        final String requestPath = request.getPathInfo();
+        LOGGER.log(Level.INFO, "GET {0}", requestPath);
```

Comment:
Sonar is flagging a vulnerability here. Do we reckon some sanitization is required on the parameter here or do we reckon `getPathInfo` shouldn't contain anything of concern?

Reply:
Thanks for the interest @antares1470 in this. I saw this also and its one of the reasons why I've kept this as a draft. My thinking was that it would be OK but I was looking to do some research to double check.